smartconfig/exceptions.py

```diff
@@ -0,0 +1,27 @@
+class SmartconfigException(Exception):
+    """Base exception of `smartconfig` library."""
+    pass
+
+
+class PathConflict(ValueError, SmartconfigException):
+    """
+    Two configuration entries point to the same path.
+
+    Provide a `path` argument to the metaclass to bypass this restriction.
+    """
+    pass
+
+
+class ConfigurationKeyError(AttributeError, SmartconfigException):
+    """An invalid key name has been used."""
+    pass
+
+
+class InvalidOperation(RuntimeError, SmartconfigException):
+    """An invalid operation has been performed."""
+    pass
```

Comment:
This is currently only being used to raise an error for circular references, so we could rename this to specifically be for that use case unless you foresee other uses for this exception.

Reply:
This is currently also used if an instance of `ConfigEntry` is created, although I don't mind introducing new exceptions.

https://github.com/python-discord/smartconfig/blob/d7cffa651dda055f6d399ea32aef6598568a5924/smartconfig/entry.py#L123

Reply:
Oh, I missed that. I feel like a circular reference is more of a configuration error anyway so it probably doesn't need its own exception type.

Reply:
Changed the exception type used in 56ed6b4